src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -426,25 +601,40 @@ export class SavedObjectsRepository {
     }
 
     const { refresh = DEFAULT_REFRESH_SETTING } = options;
-
     const allTypes = Object.keys(getRootPropertiesObjects(this._mappings));
+    const typesToUpdate = allTypes.filter(type => !this._registry.isNamespaceAgnostic(type));
 
-    const typesToDelete = allTypes.filter(type => !this._registry.isNamespaceAgnostic(type));
-
-    const esOptions = {
-      index: this.getIndicesForTypes(typesToDelete),
+    const updateOptions = {
+      index: this.getIndicesForTypes(typesToUpdate),
       ignore: [404],
       refresh,
       body: {
+        script: {
+          source: `
+              if (!ctx._source.containsKey('namespaces')) {
```

Comment:
Apologies for sending you down the path of trying to use inline scripts @jportner without realizing the caveats that @legrego brought up.

In an ideal world, where engineering effort could be completely ignored, we'd use the approach that takes advantage of inline scripts when they were enabled and fall-back to a less performant version when they weren't. 

I've tried to think of a way to prevent having to do a series of batched `_search` operations and subsequent update/delete options,  but I haven't been able to...

Reply:
I added 9f9effb to help address this concern.